app/controllers/katello/api/v2/alternate_content_sources_controller.rb

```diff
@@ -0,0 +1,107 @@
+module Katello
+  class Api::V2::AlternateContentSourcesController < Api::V2::ApiController
+    include Katello::Concerns::FilteredAutoCompleteSearch
+
+    before_action :find_authorized_katello_resource, :only => [:show, :update, :destroy]
+    before_action :find_smart_proxies
+
+    def_param_group :acs do
+      param :name, String, desc: N_("Name of the alternate content source"), required: false
+      param :description, String, desc: N_("Description for the alternate content source")
+      param :base_url, String, desc: N_('Base URL for finding alternate content')
+      param :subpaths, Array, desc: N_('Path suffixes for finding alternate content')
+      param :smart_proxy_ids, Array, desc: N_("Ids of smart proxies to associate"), required: false
+      param :content_type, RepositoryTypeManager.defined_repository_types.keys, desc: N_("The content type for the Alternate Content Source")
+      param :alternate_content_source_type, AlternateContentSource::ALLOWED_TYPES, desc: N_("The Alternate Content Source type")
+      param :upstream_username, String, desc: N_("Basic authentication username"), required: false
+      param :upstream_password, String, desc: N_("Basic authentication password"), required: false
+      param :ssl_ca_cert_id, :number, desc: N_("Identifier of the content credential containing the SSL CA Cert"), required: false
+      param :ssl_client_cert_id, :number, desc: N_("Identifier of the content credential containing the SSL Client Cert"), required: false
+      param :ssl_client_key_id, :number, desc: N_("Identifier of the content credential containing the SSL Client Key"), required: false
+      param :http_proxy_id, :number, desc: N_("ID of a HTTP Proxy"), required: false
+      param :verify_ssl, :bool, desc: N_("If SSL should be verified for the upstream URL"), required: false
+    end
+
+    api :GET, "/alternate_content_sources", N_("List of alternate_content_sources")
+    param :label, String, desc: N_("Label of the alternate content source"), required: false
+    param :name, String, desc: N_("Name of the alternate content source"), required: false
+    param :description, String, desc: N_("Description for the alternate content source")
+    param :base_url, String, desc: N_('Base URL for finding alternate content')
+    param :subpaths, Array, desc: N_('Path suffixes for finding alternate content')
+    param :smart_proxy_ids, Array, desc: N_("Ids of smart proxies to associate"), required: false
+    param :content_type, RepositoryTypeManager.defined_repository_types.keys, desc: N_("The content type for the Alternate Content Source")
+    param :alternate_content_source_type, AlternateContentSource::ALLOWED_TYPES, desc: N_("The Alternate Content Source type")
+    param :upstream_username, String, desc: N_("Basic authentication username"), required: false
```

Comment:
We don't normally support filtering all of these options when listing a resource, but instead leverage search to do that. Any particular use case to have them all here?

Reply:
No use, I meant to take out all the params that I don't mention in index_relation.